Sources/UnleashProxyClientSwift/Poller.swift

```diff
@@ -47,11 +47,14 @@ public class Poller {
     public func stop() -> Void {
         self.timer?.invalidate()
     }
-    
-    func formatURL(context: [String: String]) -> String {
-        let params = context.keys.map({ "\($0)=\(unwrap(context[$0]))" })
 
-        return unleashUrl + "?" + params.joined(separator: "&")
+    func formatURL(context: [String: String]) -> URL? {
+        var components = URLComponents(url: unleashUrl, resolvingAgainstBaseURL: false)
+        components?.queryItems = context.map { key, value in
```

Comment:
safe URL manipulation instead of string concatenation. No matter what strings we put into context the URL is always correctly escaped